### ISSUE TITLE
ECONNRESET issue again 

### ISSUE THREAD

**Issue by ecoruh** [squaremo/amqp.node#227]
4 months on I have identical symptoms reported in now closed issue #188. 

I have node 5.5.0, RMQ 3.6.0 Erlang 18:2 on Ubuntu Trusty 14.04.3 LTS. 

This sanity test works: 

`openssl s_client -connect localhost:5671`

However I have  the same problem when I run my node client app:  
`{ [Error: read ECONNRESET] code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }`

RMQ ubuntu_sasl log:

```
=CRASH REPORT==== 28-Jan-2016::21:13:52 ===
  crasher:
    initial call: tls_connection:init/1
    pid: <0.755.0>
    registered_name: []
    exception exit: {function_clause,
                        [{tls_v1,enum_to_oid,
                             [28],
                             [{file,"tls_v1.erl"},{line,404}]},
                         {ssl_handshake,
```
The client code:

```javascript
var amqp = require('amqplib');
var fs = require('fs');

var opts = {
  cert: fs.readFileSync('/root/client/cert.pem'),
  key: fs.readFileSync('/root/client/key.pem'),
  passphrase: 'password',
  ca: [fs.readFileSync('/root/testca/cacert.pem')]
};
// var opts = {};

var open = amqp.connect('amqps://localhost', opts);

open.then(function(conn) {
  process.on('SIGINT', conn.close.bind(conn));
  return conn.createChannel().then(function(ch) {
    ch.sendToQueue('foo', new Buffer('Hello World!'));
  });
}).then(null, console.warn);
```
Any help is appreciated, I am stuck.

**Comment by michaelklishin** [comment]
@ecoruh #222.

**Comment by ecoruh** [comment]
@michaelklishin I have the latest of everything. I upgraded Erlang to 18.2. Why do you think it is not working?

**Comment by michaelklishin** [comment]
@ecoruh the TLS handshake exception posted is the same we've seen in R16B03 releases but never with later versions. Make sure you actually run the Erlang version you think you do. `rabbitmqctl status` mentions it, for example.

Other clients can work because they default to a different TLS version or even SSLv3.

**Comment by ecoruh** [comment]
Thank @michaelklishin I really appreciate your help. I checked `rabbitmqctl status`, you are right, indeed it shows R16B03, even though I thought I installed 18.2!
```
rabbitmq status
...
{erlang_version,
     "Erlang R16B03 (erts-5.10.4) [source] [64-bit] [smp:2:2] [async-threads:64] [kernel-poll:true]\n"},
```
However I installed Erlang version 18.2. Here is the proof: 
```
root@ubuntu:/etc/app# erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().'  -noshell
"18"
```
The question is how can I get RabbitMQ use the new version? Thanks again.

**Comment by michaelklishin** [comment]
RabbitMQ uses `erl` in PATH, that's it.

**Comment by michaelklishin** [comment]
RabbitMQ has an OS X build which bundles Erlang, that package is used by Homebrew. It currently uses Erlang 17.4 IIRC. So if you are running that standalone build, it needs to be upgraded.

**Comment by ecoruh** [comment]
I am working on Ubuntu VM. I followed this bash script (after modifying to use the right version), to install Erlang 18.2: https://gist.github.com/bryanhunter/10380945

**Comment by ecoruh** [comment]
Actually erl shows version 18. i don't get it:
```
root@ubuntu:/etc/app# erl
Erlang/OTP 18 [erts-7.2] [source] [64-bit] [smp:2:2] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V7.2  (abort with ^G)
```

**Comment by ecoruh** [comment]
@michaelklishin I failed to get RabbitMQ point to new erlang version 18.2.
Proof of which erl PATH refers to. It is 18.2:
```
root@ubuntu:/etc/app# which erl
/usr/local/bin/erl
root@ubuntu:/etc/app# ls -la /usr/local/bin/erl
lrwxrwxrwx 1 root root 29 Jan 30 10:11 /usr/local/bin/erl -> /usr/local/lib/erlang/bin/erl
root@ubuntu:/etc/app# erl
Erlang/OTP 18 [erts-7.2] [source] [64-bit] [smp:2:2] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V7.2  (abort with ^G)
1> 
BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
       (v)ersion (k)ill (D)b-tables (d)istribution
^Croot@ubuntu:/etc/app# 
```
Proof of what RabbitMQ thinks erl version is (16.03):
```
{erlang_version,
     "Erlang R16B03 (erts-5.10.4) [source] [64-bit] [smp:2:2] [async-threads:64] [kernel-poll:true]\n"},
```
Stuck again.

**Comment by ecoruh** [comment]
@michaelklishin I am glad to say my issue is resolved. My client/server hello-world app using TLS now works. 

My bad was I downloaded the src and built erlang myself, the script that I downloaded did not install it properly. Whereas after I followed the following procedure erlang 18.2 was installed properly and recognised by RabbitMQ:

http://stackoverflow.com/questions/27215936/how-to-upgrade-the-erlang-version-on-ubuntu-r14b04-to-latest-stable-version

Thank you very much for your help, without you I wouldn't be able to sort it out myself. I appreciate your time, goodwill and professionalism. Thank you.

I am now closing this issue. Thanks.
